### PR TITLE
Add e2e test for static JSON files

### PR DESCRIPTION
**Pull request by ghengeveld**
Issue: -

## What I did

Added an e2e test suite for static JSON files. Currently only `index.json` is tested. I intend to add one for `modules.json` in #19228.

## How to test

Just run the e2e tests.

### CODE REVIEW

File: code/e2e-tests/json-files.spec.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  test('should have index.json', async ({ page }) => {
+    // eslint-disable-next-line jest/valid-expect-in-promise
+    const json = await page.evaluate(() => fetch('/index.json').then((res) => res.json()));
+
+    expect(json).toEqual({
+      v: expect.any(Number),
+      entries: expect.objectContaining({
+        'example-button--primary': {
+          id: 'example-button--primary',
+          importPath: expect.stringMatching(/\/Button\.stories\.[tj]sx$/),
```
Comment (tmeasday) [inline code/e2e-tests/json-files.spec.ts:20]:
```suggestion
          importPath: expect.stringMatching(/\/Button\.stories\.[tj]sx?$/),
```

Comment (tmeasday) [reply code/e2e-tests/json-files.spec.ts]: Not all stories use JSX

**Review by tmeasday — APPROVED** [review]
LGTM. Tests were failing but I think I fixed.

File: code/e2e-tests/json-files.spec.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import process from 'process';
+
+const storybookUrl = process.env.STORYBOOK_URL || 'http://localhost:8001';
+
+test.describe('JSON files', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(storybookUrl);
+  });
+
+  test('should have index.json', async ({ page }) => {
```
Comment (yannbf) [inline code/e2e-tests/json-files.spec.ts:11]: would you mind adding one extra case for `stories.json`? Should behave just the same

Comment (ndelangen) [reply code/e2e-tests/json-files.spec.ts:11]: if that is required, let's do that in another PR 👍

**Review by ndelangen — APPROVED** [review]

**Pull request merged**